dbeam-core/src/main/java/com/spotify/dbeam/args/SqlQueryWrapper.java

```diff
@@ -0,0 +1,129 @@
+/*-
+ * -\-\-
+ * DBeam Core
+ * --
+ * Copyright (C) 2016 - 2019 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.dbeam.args;
+
+import java.io.Serializable;
+
+/**
+ * Wrapper class for raw SQL query (SELECT statement).
+ */
+public class SqlQueryWrapper implements Serializable {
```

Comment:
Generic Comment, since this PR changes the way the query is being constructed:

Would a query builder pattern make sense?
I am thinking:
`DbeamQueryBuilder` as a class,
`fromTable` and `fromQueryFile` as Smart Constructors.

`withLimit`, `withSplitColumn`, `withPartitionColumn` as modifiers for the query.

`getExtractQueries`, and `getMinMaxSplitQuery` to trigger building the actual query and returning a String.

The primary aim is to have a better way to compose complex queries by adding things together.

In the implementation, every query modifier, forms a new sub query. 
```
val source = OneOf(userProvidedQuery, tableName, currentQueryState) // delegated logic for finding out the source.
```
`withPartitionColumnModifier` means `SELECT * FROM ($source) s WHERE $partitionColumnModifier`
`with....` means `SELECT * FROM  ($source) s WHERE $....`

the `source` is a disjunction which can itself be another query. So the final query would look like:
```
SELECT * FROM (
  SELECT * FROM (
    user_provided_query
  ) x WHERE partitionColumn > '....' 
) y WHERE min(splitColumn) > ... and max(splitColumn) <... 
```

Adding a query modifier, puts the previous query in a sub query and queries from it.
This would allow a more structured way to manage modifiers and sources.


Another way might be having CTEs.
Since we are dealing with rDBMS here, I think it is safe to assume that query planners would push down the predicates as needed.

Reply:
@anish749 I addressed some the comments.